staging/src/k8s.io/client-go/rest/transport.go

```diff
@@ -112,7 +112,7 @@ func (c *Config) TransportConfig() (*transport.Config, error) {
 	}
 
 	if c.Dial != nil {
-		conf.DialHolder = &transport.DialHolder{Dial: c.Dial}
+		conf.DialHolder = &transport.DialHolder{Dial: c.Dial, DisableCache: true}
```

Comment:
I think the problem with all of these types of changes is that they make assumptions about how the caller was using this code.  A caller of this method could be re-using the return value, meaning they would correctly be getting the TLS cache benefits, which this PR would now disable.  We allow for infinite flexibility, so every change has the potential to break someone in a subtle way.

Reply:
I agree with your perspective, and originally I wanted to expose the DialHolder on the `rest.Config` struct, however that is a larger change. It would provide a more intentional way of re-using the DialHolder when the caching behavior is expected.

As things exist now, unless the caller takes care to re-use the returned DialHolder, it's not obvious that a memory leak occurs when the `Dial` function is configured. The places we most often encounter this issue are [client.New()](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#New) and [kubernetes.NewForConfig()](https://pkg.go.dev/k8s.io/client-go@v0.30.1/kubernetes#NewForConfig), which make re-using the `DialHolder` non-obvious.